app/controllers/global_search_controller.rb

```diff
@@ -0,0 +1,14 @@
+class GlobalSearchController < ApplicationController
+
+  customer_tab :all
+  before_filter :authenticate_user!
+  before_filter :check_acting_as
+
+  def index
+    @results = {
```

Comment:
Yeah, I wasn't quite sure what I wanted to do with this, but this seemed like a reasonable balance of not over-abstracting and still leaving room for flexibility. Once we have a third thing we're searching on, we might have a better idea of how to abstract this and make it extensible to engines.